### ISSUE TITLE
"conan config install" with git URL fails if storage path is not set

### ISSUE THREAD

**Issue by jakobjw** [conan-io/conan#6950]
### Environment Details
  * Operating System+version: Windows 10
  * Conan version: 1.24.1 (did not occur in e.g. 1.11.2)
  * Python version: 3.6.5

### Steps to reproduce
* Remove storage path entry in conan.conf
* Run: "conan config install <conan_config_git_repo_url>"

### Logs
```
Trying to clone repo: <conan_config_git_repo_url>
Repo cloned!
Copying file cacert.pem to C:\PUBLIC\conan\.conan\.
Defining remotes from remotes.txt
ERROR: Value provided for package version, '.git' (type Version), is an invalid name. Valid names MUST begin with a letter, number or underscore, have between 2-51 chars, including letters, numbers, underscore, dot and dash
```

**Comment by memsharded** [comment]
Hi @jakobjw 

Lets have a look at this. Quick question, why are you removing the storage_path?

**Comment by jakobjw** [comment]
Hi @memsharded 

We have a script which automatically updates the Conan configuration to have a "fresh environment". Since we sometimes had trouble with the old settings or a "corrupted" old Conan storage, we run:
* conan config rm general
* conan config rm storage

before
* conan config install ...

**Comment by memsharded** [comment]
Thanks! I have been able to reproduce. I have checked if a fix is easy, but seems a bit complex. We might need to wait until 2.0, because right now there is an ugly fallback that returns the cache folder if the storage folder is not defined, and that is causing the error, but it is likely that if we try to remove that, it will be breaking for other users.

As overwriting the storage path variable should work always, I would suggest to do NOT remove it before the ``conan config install``, as it will overwrite its value. Please try that as a workaround, and report if that doesn't work.

**Comment by memsharded** [comment]
This is the test to reproduce:

```python
    def install_repo_without_storage_test(self):
        # https://github.com/conan-io/conan/issues/6950
        folder = self._create_profile_folder()
        with self.client.chdir(folder):
            self.client.run_command('git init .')
            self.client.run_command('git add .')
            self.client.run_command('git config user.name myname')
            self.client.run_command('git config user.email myname@mycompany.com')
            self.client.run_command('git commit -m "mymsg"')

        self.client.run("config rm storage")
        self.client.run('config install "%s/.git"' % folder)
        check_path = os.path.join(folder, ".git")
        self._check("git, %s, True, None" % check_path)
```

I have checked the code twice, and it is almost impossible to fix this without breaking other users (it would need to introduce yet another opt-in in the conan.conf first).

So as there is a solution (do not remove the storage path, it will be overwritten), I am planning this to be fixed in next major 2.0, when we can break.

**Comment by memsharded** [comment]
This is no longer an issue for 2.0, because the storage folder does fallback to a ``p`` subfolder always.
Closing this as solved in 2.0, testing welcome.
